### ISSUE TITLE
[ENHANCEMENT] Multi-output regression

### ISSUE THREAD

**Issue by vtaquet** [scikit-learn-contrib/MAPIE#97]
Following this paper : http://proceedings.mlr.press/v128/messoudi20a.html

**Comment by sdestercke** [comment]
We have a better calibrated version (using copulas to better adjust target-wise confidence levels): https://arxiv.org/abs/2101.12002. Code available here: https://github.com/M-Soundouss/CopulaConformalMTR

**Comment by gmartinonQM** [comment]
Interesting, thanks for sharing ! We will definitely have a look when we will start this development.

**Comment by sdestercke** [comment]
Hi @gmartinonQM,

Many thanks for the suggestion and for the helpful tips on how to add our method to the MAPIE library.

I may not have the time right now to do it, but I think that @M-soundouss could be equally if not more interested, with maybe a bit more time on her hand right now! There is also a priori no big issues in making it model agnostic. Two question points that directly pops up into my mind:

- Basically, what we extend is the normalized version of conformal regression scores, so my guess is that the normalizing value should also be model-agnostic in the code? 

- Would it be okay to use the copulae python library within the code?

**Comment by gmartinonQM** [comment]
Hi @sdestercke, to answer your questions : 

- yes, normalizing value should be a user choice (but you can suggest a default model to use, within sklearn preferentially)
- yes, feel free to add the copulae dependency, at least in `setup.py` (INSTALL_REQUIRES) to ensure that the dependency is resolved for the python package, and in the dev environment `environment.dev.yml` for developers

**Comment by sdestercke** [comment]
@gmartinonQM Not so far, @M-Soundouss is writing her thesis, which is time-consuming. Helping in implementing multi-output regression for MAPIE is still in our to-do list, though.

**Comment by vincentblot28** [comment]
Hi @sdestercke and @M-Soundouss, do you have any news about this issue ?

**Comment by M-Soundouss** [comment]
Hello @vincentblot28 ! I started working on it, I'll keep you updated. Thank you!

**Comment by Valentin-Laurent** [comment]
Hello @M-Soundouss, long time no see 😃
I know this is a pretty old discussion, but we're in the process of triaging/closing issues.
I believe you didn't find time to work on that?
Thank you

**Comment by Valentin-Laurent** [comment]
Closing this as MultiOutput regression is not on our roadmap.

Feel free to reopen it if you have additional insights to add to the discussion or if you'd like to contribute to MAPIE on this topic.
